### PR TITLE
bug/silent-error-suggest-select/#96

### PR DESCRIPTION
**Pull request by eisenbergrobin**
#96

### CODE REVIEW

**Comment by eisenbergrobin** [comment]
This makes sure that if the component is unmounted (for example if a backbone container deletes it), the click event will be registered but will not bubble down the nodepath.

This only occurs if the SuggestSelect is set to keep its focus (using the `focusable` prop) and if its container is set to unmount it on selection.  

PR is ready for review.

File: src/components/inputs/SuggestSelect.jsx [hunk 1]
```diff
@@ -387,6 +387,7 @@ const SuggestSelect = React.createClass({
 
   onOptionSelect(option, e) {
     this.selectOption(option.id);
+    if (!this.isMounted()) e.stopPropagation();
```
Comment (zallek) [inline src/components/inputs/SuggestSelect.jsx]: a bit worried about that
https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html

Comment (eisenbergrobin) [inline src/components/inputs/SuggestSelect.jsx]: Yep I did not know it was on its way to deprecation. 

I'll change this PR to use a `mounted` attribute on the class, it'll be cleaner that way.

Comment (eisenbergrobin) [inline src/components/inputs/SuggestSelect.jsx]: Done: we now use a `_mounted` attribute, as specified in the blog post. 

I still don't understand why `react` doesn't differentiate between an internal (private) `_isMounted` used by React and an external (public) `isMounted` that does what I've done here (keeping track of mount state using react lifecycle callbacks). 

In any case as the blog post says, the `isMounted` method is no longer available in ES6 React components, so this is definitely something we don't want to keep doing.

Comment (zallek) [inline src/components/inputs/SuggestSelect.jsx]: you still didn't turned to this._mounted

**Comment by eisenbergrobin** [comment]
My bad, sorry, I went too fast. 

Fixed.

**Pull request merged**